core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -73,17 +77,25 @@ class BrokerMetadataListener(brokerId: Int,
   /**
    * Handle new metadata records.
    */
-  override def handleCommits(lastOffset: Long, records: util.List[ApiMessage]): Unit = {
-    eventQueue.append(new HandleCommitsEvent(lastOffset, records))
+  override def handleCommit(reader: BatchReader[ApiMessageAndVersion]): Unit = {
+    try {
+      while (reader.hasNext) {
+        val batch = reader.next()
```

Comment:
Yes, you are right. Let me see if I can take this a little further.

Reply:
By the way, I am okay not fixing it in this PR since it is not a regression. If we decide to do that, then let's update the PR description and file an issue for the broker and controller implementation.

Reply:
I went ahead and did it here since it seemed straightforward.